pandas/core/frame.py

```diff
@@ -217,6 +217,7 @@ def __init__(self, data=None, index=None, columns=None, dtype=None,
             dtype = self._validate_dtype(dtype)
 
         if isinstance(data, DataFrame):
+            data = data.iloc[:,:]
             data = data._data
```

Comment:
you need to define ``BlockManager.view()`` this will do all of the heavy lifting, everything else will flow from there